### PR TITLE
IDENTITY-5983: Fix user password reset after tenant is deactivated

### PR DESCRIPTION
**Pull request by emswbandara**

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java [hunk 1]
```diff
@@ -491,7 +491,10 @@ private ClientResponse getAPIResponse(String path, String method, List<Pair> que
 
         statusCode = response.getStatusInfo().getStatusCode();
         responseHeaders = response.getHeaders();
-
+        String responseBody = null;
+        if (response.hasEntity()) {
```
Comment (IsuraD) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: why this change was needed.?

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java [hunk 2]
```diff
@@ -190,6 +191,19 @@ public void setPasswordPost(ResetPasswordRequest resetPasswordRequest) throws Ap
...
         }
 
+        String userTenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        List<Property> properties = resetPasswordRequest.getProperties();
+        for(Property property : properties){
+            if(StringUtils.equalsIgnoreCase("tenantdomain", property.getKey())){
+                userTenantDomain = property.getValue();
+            }
+        }
+
```
Comment (IsuraD) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: remove the userTenantDomain property after identifying it from properties

Comment (emswbandara) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: updated

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java (hunk 1 above)
Comment (emswbandara) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/ApiClient.java]: to avoid java.io.IOException: stream is closed exception which occurred with the previous approach

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java [hunk 3]
```diff
@@ -190,6 +191,21 @@ public void setPasswordPost(ResetPasswordRequest resetPasswordRequest) throws Ap
             throw new ApiException(400, "Missing the required parameter 'resetPasswordRequest' when calling setPasswordPost");
         }
 
+        String userTenantDomain = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+        List<Property> properties = resetPasswordRequest.getProperties();
+        for(Property property : properties){
```
Comment (madurangasiriwardena) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: Formatting issue

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp [hunk 4]
```diff
@@ -24,6 +24,11 @@
 <%
     String confirmationKey = request.getParameter("confirmation");
     String callback = request.getParameter("callback");
+    String tenantDomain = request.getParameter(IdentityManagementEndpointConstants.TENANT_DOMAIN);
+
+    if(StringUtils.isBlank(tenantDomain)) {
```
Comment (madurangasiriwardena) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp]: Formatting issue

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java [hunk 5]
```diff
@@ -190,6 +191,21 @@ public void setPasswordPost(ResetPasswordRequest resetPasswordRequest) throws Ap
...
+            if(StringUtils.equalsIgnoreCase(IdentityManagementEndpointConstants.TENANT_DOMAIN, property.getKey())){
```
Comment (pulasthi7) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: Fix formatting

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp (hunk 4 above)
Comment (pulasthi7) [inline components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp]: Fix formatting

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java (hunk 3 above)
Comment (emswbandara) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: Fixed with #1300

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp (hunk 4 above)
Comment (emswbandara) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp]: Fixed with #1300

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java (hunk 5 above)
Comment (emswbandara) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/java/org/wso2/carbon/identity/mgt/endpoint/client/api/NotificationApi.java]: Fixed with #1300

File: components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp (hunk 4 above)
Comment (emswbandara) [reply components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint/src/main/webapp/password-recovery-confirm.jsp]: Fixed with #1300
